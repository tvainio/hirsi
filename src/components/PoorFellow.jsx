import React from "react";
import PropTypes from "prop-types";
import { Panel } from "react-bootstrap";

class PoorFellow extends React.Component {
  constructor(props) {
    super(props);
    this.state = {};
  }
  render() {
 //   let { timestamp, id, ttl, ...data } = this.props.data;
    const TOTAL_STATES = 11;
    return (
      <div>
        <Panel>
          <Panel.Body>
            Poor Fellow! Always in trouble...
            <div id="state">game state {this.props.gameState+"/"+TOTAL_STATES}</div>
          </Panel.Body>
        </Panel>
      </div>
    );
  }
}

PoorFellow.propTypes = {
  gameState: PropTypes.number.isRequired
};

export default PoorFellow;
