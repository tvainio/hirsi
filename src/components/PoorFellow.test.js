import React from "react";
import { shallow } from "enzyme";
import PoorFellow from "./PoorFellow";
import renderer from "react-test-renderer";

it("matches snapshot", () => {
  const tree = renderer
    .create(<PoorFellow gameState={2}/>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it("shows gamestate", () => {
  const state = 3;
  const wrapper = shallow(
    <PoorFellow gameState={state} />
  );
  const temp = wrapper.find("#state");
  expect(temp.text()).toContain("game state 3/11");
});
