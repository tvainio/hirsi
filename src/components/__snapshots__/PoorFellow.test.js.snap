// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div>
  <div
    className="panel panel-default"
  >
    <div
      className="panel-body"
    >
      Poor Fellow! Always in trouble...
      <div
        id="state"
      >
        game state 
        2/11
      </div>
    </div>
  </div>
</div>
`;
